### PR TITLE
gamenet: Add 0.7 connless inf3 (closed #77)

### LINKED ISSUE #77: gamenet: missing connless info (inf3)
https://github.com/heinrich5991/libtw2/blob/7885c99974ee445ce13297b72ae3e7c6ea3b969d/gamenet/generate/spec/teeworlds-0.7.5.json#L1127-L1133

there is "request info" which maps to GETINFO here

https://github.com/teeworlds/teeworlds/blob/26d24ec061d44e6084b2d77a9b8a0a48e354eba6/src/mastersrv/mastersrv.h#L27

but the "info" one is missing

### PR DESCRIPTION
**Pull request by ChillerDragon**
Generated using

```bash
cd gamenet/generate
python3 serialize.py --version 0.7 raw/teeworlds-0.7.5.py > spec/teeworlds-0.7.5.json
```

### CODE REVIEW

**Comment by ChillerDragon** [comment]
Changes to 0.6 are the IntStrins got changed to regular ints. And the [hostname](https://github.com/teeworlds/teeworlds/commit/57e28e5c88c52a621b2e1c7b24d9f9f9987299d9#diff-65908fcaf323f624c0a8e584e92664d0d49640bb139729c6e22a14a5b2170c3fR1092) field was added.

How it was in 0.6
https://github.com/teeworlds/teeworlds/blob/93f5bf632a3859e97d527fc93a26b6dced767fbc/src/engine/server/server.cpp#L1082-L1145

How it is in 0.7
https://github.com/teeworlds/teeworlds/blob/26d24ec061d44e6084b2d77a9b8a0a48e354eba6/src/engine/server/server.cpp#L1091-L1149

I did not touch the client list since from what I understood its fields like name, clan, score are not implemented for 0.6 either.
But the fields country, score, is player also changed from IntString to regular int

The hostname field is unpacked by the client like this. So I assume the NetStringStrict is fitting. The client defaulting it to m_aAddress is probably irrelevant.

```C++
str_copy(pInfo->m_aName, pUnpacker->GetString(CUnpacker::SANITIZE_CC|CUnpacker::SKIP_START_WHITESPACES), sizeof(pInfo->m_aName));
str_clean_whitespaces(pInfo->m_aName);
str_copy(pInfo->m_aHostname, pUnpacker->GetString(CUnpacker::SANITIZE_CC|CUnpacker::SKIP_START_WHITESPACES), sizeof(pInfo->m_aHostname));
if(pInfo->m_aHostname[0] == 0)
	str_copy(pInfo->m_aHostname, pInfo->m_aAddress, sizeof(pInfo->m_aHostname));
```

The num_client fields could be capped to MAX_CLIENTS but I think it would be nicer if there is space for 128+ servers in this spec.

**Comment by ChillerDragon** [comment]
Current state untested. I was hoping to verify it with the dissector but it does not seem to work yet.

File: gamenet/generate/loader.py [hunk 1]
```diff
@@ -180,6 +182,24 @@ def fix_network(network, version):
                     NetClients("clients"),
                 ]),
             ]
+        if version in (VERSION_0_7):
+            network.Connless += [
+                NetConnless("Info", "inf3", [
+                    NetIntString("token"),
```
Comment (heinrich5991) [inline gamenet/generate/loader.py:188]: Is this really `NetIntString`?

**Comment by ChillerDragon** [comment]
Ah nice seems like the token fixed the dissector looks kinda correct now

image

File: gamenet/generate/loader.py [hunk 2]
```diff
@@ -180,6 +182,24 @@ def fix_network(network, version):
                     NetClients("clients"),
                 ]),
             ]
+        if version == VERSION_0_7:
+            network.Connless += [
+                NetConnless("Info", "inf3", [
+                    NetU8("token"),
```
Comment (heinrich5991) [inline gamenet/generate/loader.py:188]: Should be a `NetIntAny`, no? It seems to leak into the `version` string in your screenshot.

Comment (ChillerDragon) [reply gamenet/generate/loader.py]: image
yes looks much better :sweat_smile:

**Pull request merged**